plone/scale/tests/test_scale.py

```diff
@@ -114,6 +116,40 @@ def testAutomaticGreyscale(self):
         self.assertEqual(image.mode, "L")
         self.assertEqual(image.format, "JPEG")
 
+    def testMaintainLAModeForPalettePNG(self):
+        """Test that a palette (P) mode PNG with transparency maintains alpha channel."""
+        # Open the image using the project's pattern
+
+        # Verify the original image is in P mode
+        original_image = PIL.Image.open(StringIO(GREYSCALE_IMG))
+        self.assertEqual(original_image.mode, "P", "Test image should start in P mode")
+
+        # Scale the image
+        scaled_data, format_, size = scaleImage(GREYSCALE_IMG, 100, None, "contain")
+
+        # Open the scaled result and check properties
+        scaled_image = PIL.Image.open(StringIO(scaled_data))
+
+        # Check the image is not all black (not losing transparency)
+        histogram = scaled_image.histogram()
+        self.assertFalse(
+            histogram[0] > sum(histogram[1:]),
+            "Scaled image appears to be mostly black, which indicates alpha information was lost",
+        )
+
+        # Verify transparency is maintained in some form
+        has_transparency = scaled_image.mode == "LA"
```

Comment:
The variable `has_transparency` is not required. Use `assertEqual`.

Reply:
Done!